fragment.go

```diff
@@ -1425,6 +1419,50 @@ func (f *fragment) importValue(columnIDs, values []uint64, bitDepth uint) error
 	return nil
 }
 
+// importRoaringBytes imports from standard roaring data format defined at
+// https://github.com/RoaringBitmap/RoaringFormatSpec or from pilosa's version
+// of the roaring format.
+func (f *fragment) importRoaringBytes(roaringBytes []byte) error {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	bm := roaring.NewBitmap()
+	err := bm.UnmarshalBinary(roaringBytes)
+	if err != nil {
+		return err
+	}
+
+	// get a list of keys in order to update the cache
+	i, _ := bm.Containers.Iterator(0)
+	set := make([]uint64, 0)
+	var lastRow uint64 = math.MaxUint64
+
+	for i.Next() {
+		key, _ := i.Value()
+
+		// virtual row for the current container
+		vRow := key >> containersPerRowSegment
+
+		// skip dups
+		if vRow == lastRow {
+			continue
+		}
+			set = append(set, vRow)
+		lastRow = vRow
+	}
+
+	if f.storage.Count() > 0 {
+		bm = f.storage.Union(bm)
+	}
+
+	for _,rowID := range set {
+		n := bm.CountRange(rowID*ShardWidth, (rowID+1)*ShardWidth)
+		f.cache.BulkAdd(rowID, n)
+	}
+
```

Comment:
@tgruben do we need to invalidate the cache here?

Reply:
most definately